online/src/app/products-list/products-list.component.ts

```diff
@@ -0,0 +1,37 @@
+import { Component, OnInit } from '@angular/core';
+import { ProductViewModel } from '../products';
+import { AuthentificationService } from '../services/authentification.service';
+import { ProductService } from '../services/product.service';
+import { admin, customer } from '../utils';
+
+@Component({
+  selector: 'app-products-list',
+  templateUrl: './products-list.component.html',
+  styleUrls: ['./products-list.component.scss']
+})
+export class ProductsListComponent implements OnInit {
+  productID: number = -1;
+  products: ProductViewModel[] | undefined;
+  hasCustomerRole = this.authService.hasRoleType(customer);
+  hasAdminRole = this.authService.hasRoleType(admin);
+  
+  constructor(private productService: ProductService, private authService: AuthentificationService) { }
+
+
+  ngOnInit(): void {
+    this.getProductsList();
+  }
+
+  getProductsList(): void {
+    this.productService.getProducts().subscribe(products => this.products = products);
```

Comment:
I recommend splitting this component into two and passing getProducts with | async pipe, such that you have its latest value in the child component.